wrapanapi/systems/redfish.py

```diff
@@ -33,5 +71,80 @@ def _identifying_attrs(self):
     def info(self):
         return 'RedfishSystem url={}'.format(self.url)
 
-    def __del__(self):
-        """Disconnect from the API when the object is deleted"""
+    def server_stats(self, *requested_stats, **kwargs):
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
+
+        # Retrieve and return the stats
+        requested_stats = requested_stats or self._stats_available
+
+        return {stat: self._server_stats_available[stat](self, requester)
+                for stat in requested_stats}
+
+    def server_inventory(self, *requested_items, **kwargs):
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
+
+        # Retrieve and return the inventory
+        requested_items = requested_items or self._server_inventory_available
+
+        return {item: self._server_inventory_available[item](self, requester)
+                for item in requested_items}
+
+    def get_server(self, resource_id):
+        """
+        Fetch a Redfish resource that represents a particular physical server.
+
+        Args:
+          resource_id: the Redfish @odata.id of the resource representing the
+             server to be retrieved
+        """
+        return self.api_client.find(resource_id)
+
+    def num_servers(self):
+        return len(self.api_client.Systems.Members)
+
+    def get_server_cores(self, resource_id):
```

Comment:
All of these methods take a resource_id in order to identify the server - when there's already a get_server method.

I think that you should implement a RedfishServer class that extents wrapanapi.entity class (like VMs, templates for other providers)

This way `get_server` would return a `RedfishServer` object, that then contains the `get_cores`, `get_memory`, etc methods. They can all reference `self.raw`, which should be the RedFish api_client object with the raw parameters.

The pattern described above is one of the core design principles of wrapanapi 3.0.

Reply:
Here is where the `entities.server.Server` class would come in handy, with a specific `RedfishServer` defined locally.